language-examples/remedial-js/8.0-JS-101/README.md

```diff
@@ -0,0 +1,307 @@
+# 8.0 - JS101
+
+# Purpose
+
+JavaScript is an untyped, interpreted programming language that can accommodate
+a variety of  programming paradigms. Among other things, a lot of modern
+JavaScript code heavily leverages functional programming style. The
+combination of weak typing and functional methods can make JavaScript code a
+bit hard to understand for those coming from strongly typed object-oriented
+languages such as  Java.
+
+This module is intended for an audience who is new to JavaScript, or one that
+simply needs a refresher. We will walk through the basic JavaScript
+constructs.
+
+
+#Function Chaining / Fluent interfaces
+
+If we look at files that make up an AngularJS application, a typical file
+would often consist of a single giant JavaScript statement where multiple
+method calls are chained together: we call a method on an object, get another
+object, then call a method on that object, get another method, etc. JavaScript
+allows us to insert white space before the `.` that precedes a method
+invocation
+
+```javascript
+/** Example of method chaining in jQuery */
+
+
+//////////////////////////
+// WITHOUT METHOD CHAINING
+
+var $div = $('#my-div');         // assign to var
+
+$div.css('background', 'blue');  // set BG
+$div.height(100);                // set height
+$div.fadeIn(200);                // show element
+
+
+///////////////////////
+// WITH METHOD CHAINING
+
+$('#my-div').css('background', 'blue').height(100).fadeIn(200);
+
+// often broken to multiple lines:
+$('#my-div')
+  .css('background', 'blue')
+  .height(100)
+  .fadeIn(200);
+```
+
+is really equivalent to this:
+
+```javascript
+  angular.module('ngcourse').controller(...);
+```
+
+# Exercise 1
+
+To support this `fluent` programming style, Lets define our own class and add method chaining.  Notice each prototype method is returning an object.
+This code is running live in the exercise.js file, so you can open the browser, navigate to this exercise, and see the code in action in the console.
+
+```javascript
+// define the class
+var Kitten = function() {
+  this.name = 'Garfield';
+  this.color = 'brown';
+  this.gender = 'male';
+};
+
+Kitten.prototype.setName = function(name) {
```

Comment:
I think if you're going to use prototype methods for your example you should explain a little bit about the prototype chain first. It may be worth going into how JavaScript methods can be defined in different ways, and what the effects of that is.